includes/payment-methods/class-wc-stripe-express-checkout-element.php

```diff
@@ -189,20 +189,21 @@ public function javascript_params() {
 				'clear_cart'                => wp_create_nonce( 'wc-stripe-clear-cart' ),
 				'pay_for_order'             => wp_create_nonce( 'wc-stripe-pay-for-order' ),
 			],
-			'i18n'               => [
+			'i18n'                   => [
 				'no_prepaid_card'  => __( 'Sorry, we\'re not accepting prepaid cards at this time.', 'woocommerce-gateway-stripe' ),
 				/* translators: Do not translate the [option] placeholder */
 				'unknown_shipping' => __( 'Unknown shipping option "[option]".', 'woocommerce-gateway-stripe' ),
 			],
-			'checkout'           => $this->express_checkout_helper->get_checkout_data(),
-			'button'             => $this->express_checkout_helper->get_button_settings(),
-			'is_pay_for_order'   => $this->express_checkout_helper->is_pay_for_order_page(),
-			'has_block'          => has_block( 'woocommerce/cart' ) || has_block( 'woocommerce/checkout' ),
-			'login_confirmation' => $this->express_checkout_helper->get_login_confirmation_settings(),
-			'is_product_page'    => $this->express_checkout_helper->is_product(),
-			'is_checkout_page'   => $this->express_checkout_helper->is_checkout(),
-			'product'            => $this->express_checkout_helper->get_product_data(),
-			'is_cart_page'       => is_cart(),
+			'checkout'               => $this->express_checkout_helper->get_checkout_data(),
+			'button'                 => $this->express_checkout_helper->get_button_settings(),
+			'is_pay_for_order'       => $this->express_checkout_helper->is_pay_for_order_page(),
+			'has_block'              => has_block( 'woocommerce/cart' ) || has_block( 'woocommerce/checkout' ),
+			'login_confirmation'     => $this->express_checkout_helper->get_login_confirmation_settings(),
+			'is_product_page'        => $this->express_checkout_helper->is_product(),
+			'is_checkout_page'       => $this->express_checkout_helper->is_checkout(),
+			'product'                => $this->express_checkout_helper->get_product_data(),
+			'is_cart_page'           => is_cart(),
+			'taxes_based_on_billing' => get_option( 'woocommerce_tax_based_on' ) === 'billing',
```

Comment:
```suggestion
			'taxes_based_on_billing' => wc_tax_enabled() && get_option( 'woocommerce_tax_based_on' ) === 'billing',
```

To cover cases where the merchant later disables taxes.